packages/hebao_v1/contracts/modules/core/ForwarderModule.sol

```diff
@@ -222,6 +215,28 @@ abstract contract ForwarderModule is BaseModule
         );
     }
 
+    function batchCall(
+        address   wallet,
+        address[] calldata to,
+        bytes[]   calldata data
+        )
+        external
+        txAwareHashNotAllowed()
+        onlyFromWalletOrOwnerWhenUnlocked(wallet)
+    {
+        require(to.length == data.length, "INVALID_DATA");
+
+        for (uint i = 0; i < to.length; i++) {
+            verifyTo(to[i], wallet, data[i]);
+            // The trick is to append the really logical message sender and the
+            // transaction-aware hash to the end of the call data.
+            (bool success, ) = to[i].call(
+                abi.encodePacked(data[i], wallet, bytes32(0))
```

Comment:
Why not use `Wallet.transact(...)`

Reply:
No real reason. But this will be quite a bit cheaper because this'll call modules, and if we have to pass through the wallet that's an additional 2 contract calls we have to pass through (proxy + wallet) to get to the same destination.

How could this simplify things? Now it just does the same thing we do in normal meta-txs.

Reply:
I figured out, was going to say the same thing.